server/grpc/rpc.go

```diff
@@ -51,12 +51,44 @@ type RPC struct {
 	pipelineCount *prometheus.CounterVec
 }
 
+// Replaces legacy filter labels with new ones.
+func upconvertFilterLabels(filter rpc.Filter) rpc.Filter {
+	if value, ok := filter.Labels["repo"]; ok {
+		filter.Labels["woodpecker-ci.org/repo-full-name"] = value
+		delete(filter.Labels, "repo")
```

Comment:
Can we delete the old labels here already? The old labels can't be used by end users anymore?

Reply:
All the old labels have equivalent new ones, and the agents have the new ones, so it should be safe to upgrade the filters to use the new labels.
One things to add is maybe some logging that an agent filter was modified/migrated inplace.

Reply:
Yes but the labels can also be used in workflows https://woodpecker-ci.org/docs/usage/workflow-syntax#labels so if a user uses `hostname=xxx` there the agent also should have this label until we finally drop it?

Reply:
Regarding the sidenote: d89763ff6f8401b9c2767a0022eade0948653d21

Reply:
Thanks for writing it down, that helps me as well. Wouldn't it be better than to only prefix WP reserved labels with `woodpecker-ci.org` and leave labels that can be set/changes by the user unprefixed? Adding now an extra logic to determine which labels can be changed looks strange to me.

Reply:
I have now removed the `woodpecker-ci.org` prefix from the `platform`, `hostname` and `backend` labels because they are unmanaged